lib/src/json_schema/json_schema.dart

```diff
@@ -1492,7 +1503,7 @@ class JsonSchema {
     // TODO: add a more advanced check to find out if the $ref is local.
     // Does it have a fragment? Append the base and check if it exists in the _refMap
     // Does it have a path? Append the base and check if it exists in the _refMap
-    if (ref.scheme.isEmpty) {
+    if (ref.scheme.isEmpty && ref.path != _root._uri?.path) {
```

Comment:
Should it be `&& ref.path == this._uri.path` rather than comparing against `_root`?

Reply:
Or perhaps we should consult `_parent`?

Reply:
Ok, chatted live. This is an unrelated bugfix for ref resolution.

> There is a case in the code when the $ref with a fragment pointed to the root of the schema, it would actually think it was a new remote ref and attempt to resolve it that way.